### PR TITLE
Add slots and data related functions

### PR DESCRIPTION
**Pull request by felixlinker**

### CODE REVIEW

File: src/data/items.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                type.push(SLOT_TYPES.INTERNAL);
+            }
+            return [parseClass(item), type];
+        }
+    }
+    return [null, []];
+}
+
+export function getInternalItemInfo(item) {
+    if (item) {
```
Comment (willyb321) [inline src/data/items.js]: Better to check if not item, less nesting

File: src/data/items.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            }
+    }
+    return [null, []];
+}
+
+export function itemFitsSlot(item, ship, slot) {
+    let [itemSize, itemTypes] = getItemInfo(item);
+    let [slotSize, slotType] = getSlotInfo(ship, slot);
+    // Force cast to boolean for clean result
+    return !!(itemTypes.find(x => x === slotType) && itemSize <= slotSize);
```
Comment (willyb321) [inline src/data/items.js]: Might be better to use Boolean()

File: src/data/slots.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    if (slot) {
+        let m = slot.match(REG_INTERNAL_SLOT);
+        if (m) {
+            // Don't decrement the index for type 9 because it's internals start
+            // at 0
+            let i = - (ship === 'type_9_heavy' ? 0 : 1);
+            // Figure out whether this internal slot is a passenger slot
+            let type = typeof (coriolis.Ships[ship].slots.interal.filter(
+                slot => typeof slot === 'number' || slot.Name === 'Passenger'
+            ))[i] === 'number' ? SLOT_TYPES.INTERNAL : SLOT_TYPES.PASSENGER;
```
Comment (willyb321) [inline src/data/slots.js]: Might wanna break this out a bit tbh

File: src/data/slots.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            return [Number(m[2]), type];
+        }
+        m = slot.match(REG_MILITARY_SLOT);
+        if (m) {
+            let coriolisIndex = Number(m[1]) - 1; // Military slots start at 1
+            return [(coriolis.Ships[ship].slots.internal.filter(
+                x => typeof x === 'object' && x.Name === 'Military'
+            ))[coriolisIndex].class, SLOT_TYPES.MILITARY];
```
Comment (willyb321) [inline src/data/slots.js]: Same thing, break it out a bit

**Comment by felixlinker** [comment]
Addressed code review by rebase.

**Pull request merged**